### PR TITLE
Fixed some frontend linting issues

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

Fixed some linting issues being reported in the frontend.

## Related Tickets & Documents
#1828 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Richie Rich robot

### CODE REVIEW

File: app/javascript/.eslintrc.js [hunk 1]
```diff
@@ -22,9 +22,12 @@ module.exports = {
       },
     ],
     'import/prefer-default-export': 'off',
+    'no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
   },
   globals: {
     InstantClick: false,
     filterXSS: false,
+    Pusher: false,
```
Comment (nickytonline) [inline app/javascript/.eslintrc.js]: Some globals that were missing.

File: app/javascript/.eslintrc.js [hunk 2]
```diff
@@ -22,9 +22,12 @@ module.exports = {
       },
     ],
     'import/prefer-default-export': 'off',
+    'no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
```
Comment (nickytonline) [inline app/javascript/.eslintrc.js]: We still want the linter to complain about unused args or vars, but in the case of unused args that we might want to keep, prefix them with an `_`.

File: app/javascript/src/utils/getUserData.js [hunk 3]
```diff
@@ -1,23 +0,0 @@
-export function getUserData() {
```
Comment (nickytonline) [inline app/javascript/src/utils/getUserData.js]: Someone on the team can validate this, but the only references I saw to `getUserData` were in its own unit tests.

File: app/javascript/src/utils/pusher.js [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
 
 export default function setupPusher(key, callbackObjects) {
-  let pusher;
-  return import('pusher-js')
-    .then(({ }) => {
-    if (window.pusher) {
-      pusher = window.pusher
-    } else {
+  return import('pusher-js').then(_ => {
+    const {
```
Comment (nickytonline) [inline app/javascript/src/utils/pusher.js]: Leveraging [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) here and default values for the destructured variables.

File: app/javascript/.eslintrc.js (hunk 2 above)
Comment (nickytonline) [reply app/javascript/.eslintrc.js]: Another option is to just remove them. I'll let the team decide on that one.

**Comment by nickytonline** [comment]
Not sure why code climate is complaining about `getUserData` as I've deleted the file so there no longer is any code complexity. 🙃

File: app/javascript/.eslintrc.js (hunk 2 above)
Comment (maestromac) [reply app/javascript/.eslintrc.js]: We can keep it with the prefix.

File: app/javascript/src/utils/getUserData.js (hunk 3 above)
Comment (maestromac) [reply app/javascript/src/utils/getUserData.js]: Seems like we've been relying on `getUserDataAndCsrfToken`. Thanks for removing it!

**Comment by maestromac** [comment]
Hmm, CodeClimate acting up like that is new to me. We don't have to worry about it.

**Pull request merged**